### PR TITLE
Address case that no results were recorded in datafeed during resolut…

### PR DESCRIPTION
**Pull request by emaG3m**
…ion timeframe

### CODE REVIEW

File: contracts/Oracles/MedianPriceOracle.sol [hunk 1]
```diff
@@ -12,60 +12,96 @@ contract MedianPriceOracle is ScalarPriceOracleBase, TimedOracle {
...
   {
     medianDataFeed = TimeMedianDataFeed(_medianDataFeed);
-    medianTimeframe = _medianTimeframe;
-    resolutionDate = _resolutionDate;
+    medianStartDate = _resolutionDate.sub(medianTimeframe);
   }
 
   /**
-  * @dev consults tidbit dataFeed to calculate and set outcome
+  * @dev queries tidbit dataFeed to calculate and set outcome if resolution date has passed
```
Comment (mikec) [inline contracts/Oracles/MedianPriceOracle.sol]: from this contract's perspective, it's only dealing with `TimeMedianDataFeed` .. so might want to change the `@dev` comment to something like:

```
@dev: Gets a medianized result from the TimeMedianDataFeed contract and sets the outcome with the result.
```

Also, the required condition info can go in an `@notice`:

```
@notice: Throws if resolution date has passed.
```

File: contracts/Oracles/MedianPriceOracle.sol [hunk 2]
```diff
@@ -12,60 +12,96 @@ contract MedianPriceOracle is ScalarPriceOracleBase, TimedOracle {
...
   * @param startIndex index for first date in datafeed dates array within resolution timeframe
```
Comment (mikec) [inline contracts/Oracles/MedianPriceOracle.sol]: nitpick, but these sentences should start with a capital, like `startIndex Index for...`

File: contracts/Oracles/MedianPriceOracleFactory.sol [hunk 3]
```diff
@@ -5,7 +5,7 @@ import './IScalarPriceOracleFactory.sol';
 
 contract MedianPriceOracleFactory is IScalarPriceOracleFactory {
 
-    event MedianPriceOracleCreation(address indexed creator, IScalarPriceOracle medianPriceOracle, bytes ipfsHash, uint resolutionDate);
+    event MedianPriceOracleCreation(address indexed creator, IScalarPriceOracle medianPriceOracle, uint resolutionDate);
```
Comment (mikec) [inline contracts/Oracles/MedianPriceOracleFactory.sol]: this change should probably go in a separate commit, but don't worry about it if that's a pain to do at this point

File: contracts/Oracles/MedianPriceOracle.sol [hunk 4]
```diff
@@ -12,60 +12,96 @@ contract MedianPriceOracle is ScalarPriceOracleBase, TimedOracle {
...
+    require(_isValidRange(startIndex, endIndex));
     result = medianDataFeed.medianizeByIndices(startIndex, endIndex);
+
     _setOutcome(int(result));
   }
 
   // Internal Functions
 
-  function _isValidStartIndex(uint startIndex) internal returns (bool) {
+  function _isValidRange(uint startIndex, uint endIndex)
```
Comment (mikec) [inline contracts/Oracles/MedianPriceOracle.sol]: logic is looking good! we should definitely do another pass at reviewing and testing this block of code, since it will be so critical for resolving markets!

**Comment by mikec** [comment]
@emaG3m couple minor changes ... 👍

File: contracts/Oracles/MedianPriceOracle.sol (hunk 4 above)
Comment (emaG3m) [reply contracts/Oracles/MedianPriceOracle.sol]: 💯

File: contracts/Oracles/MedianPriceOracleFactory.sol (hunk 3 above)
Comment (emaG3m) [reply contracts/Oracles/MedianPriceOracleFactory.sol]: this got left behind when I actually made the ipfs commit
